src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -48,34 +48,32 @@ internal TrainTestData(IDataView trainSet, IDataView testSet)
 
         /// <summary>
         /// Split the dataset into the train set and test set according to the given fraction.
-        /// Respects the <paramref name="stratificationColumn"/> if provided.
+        /// Respects the <paramref name="groupPreservationColumn"/> if provided.
         /// </summary>
         /// <param name="data">The dataset to split.</param>
         /// <param name="testFraction">The fraction of data to go into the test set.</param>
-        /// <param name="stratificationColumn">Optional name of the column to use as a stratification column. If two examples share the same value of the <paramref name="stratificationColumn"/>
-        /// (if provided), they are guaranteed to appear in the same subset (train or test). Use this to make sure there is no label leakage from train to the test set.
-        /// If this optional parameter is not provided, a stratification columns will be generated, and its values will be random numbers .</param>
-        /// <param name="seed">Optional parameter used in combination with the <paramref name="stratificationColumn"/>.
-        /// If the <paramref name="stratificationColumn"/> is not provided, the random numbers generated to create it, will use this seed as value.
-        /// And if it is not provided, the default value will be used.</param>
-        public TrainTestData TrainTestSplit(IDataView data, double testFraction = 0.1, string stratificationColumn = null, uint? seed = null)
+        /// <param name="groupPreservationColumn">Name of a column to use as an ID for grouping rows. If two examples share the same value of the <paramref name="groupPreservationColumn"/>,
```

Comment:
I think you left that description from previous iteration with `idColumn`.
Do you want to rephrase it, since `ID` looks weird.

Reply:
Also ID is already a thing on IDs, naming something different that would be super confusing.

https://github.com/dotnet/machinelearning/blob/1f29d2ca32ccdb4a504cdc8de668daf91e24b448/src/Microsoft.Data.DataView/IDataView.cs#L132